Modules/Loadable/Segmentations/MRMLDM/vtkMRMLSegmentationsDisplayableManager2D.cxx

```diff
@@ -815,8 +853,20 @@ void vtkMRMLSegmentationsDisplayableManager2D::vtkInternal::UpdateDisplayNodePip
         {
         pipeline->Reslice->SetResliceTransform(pipeline->SliceToImageTransform);
         }
-      //TODO: Interpolate fractional labelmaps
-      pipeline->Reslice->SetInterpolationModeToNearestNeighbor();
+
+      // Set the interpolation mode from the InterpolationType field if it exists
+      // Default to nearest neighbor interpolation otherwise
+      vtkSmartPointer<vtkIntArray> interpolationType = vtkIntArray::SafeDownCast(
+        imageData->GetFieldData()->GetAbstractArray(vtkMRMLSegmentationsDisplayableManager2D::GetInterpolationTypeFieldName()));
+      if (interpolationType)
+        {
+        pipeline->Reslice->SetInterpolationMode(interpolationType->GetValue(0));
```

Comment:
Same as above, for the other array (simple pointer, initial value, check if there is actually a value in the array)